src/libraries/System.Security.Permissions/ref/System.Security.Permissions.cs

```diff
@@ -1128,6 +1128,9 @@ public sealed partial class PrincipalPermission : System.Security.IPermission, S
     [System.AttributeUsageAttribute(System.AttributeTargets.Class | System.AttributeTargets.Method, AllowMultiple=true, Inherited=false)]
     public sealed partial class PrincipalPermissionAttribute : System.Security.Permissions.CodeAccessSecurityAttribute
     {
+#if CAS_OBSOLETIONS
+        [System.ObsoleteAttribute("PrincipalPermissionAttribute is not honored by the runtime and must not be used.", true, DiagnosticId = "BCL0002", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
```

Comment:
Do we have the destination already created?  We should ensure this is all in place before the first Preview this is contained in is released.

Reply:
Right now it redirects to https://github.com/dotnet/runtime/issues/36972, pending final location of where the guidance doc will be.